### DISCUSSION TITLE
qspectrumanalyzer

### DISCUSSION THREAD

**Discussion by pauljackman2023** [ainfosec/FISSURE#48] [Hard Questions]
Having finaly got qspectrumanalyzer and a few other tools up and running, I am wondering how I go about importing a SigMF-Data file into Qspectrum for analysis, as all the plugins seem to be looking for hardware and I want to import a pre-recorded file.

**Comment by cpoore1** [comment]
That I don't know how to do. It could be tricky depending on the type of data you are looking at. If the data is something you would typically see from a normal SDR scenario, there are other programs you could try to import it into. I believe inspectrum has a lot of SigMF functionality as well as IQEngine. GQRX, baudline, URH, and SigDigger could also work. GNU Radio could be used as a file source and viewed with a qt/waterfall sink. You can transmit the data from a second radio into qspectrumanalyzer if you don't mind altering the data. FISSURE has a couple GNU Radio flow graphs for inspection on recorded files under the IQ Data>Inspection tab and some basic SigMF functionality for reading/viewing the .sigmf-meta files when loading a .sigmf-data file in the IQ Viewer. 

You could also look into the qspectrumanalyzer source code if it's really necessary or look for forks that have that functionality. The forks usually require specific dependencies outside of what normally goes with qspectrumanalyzer and could interfere with other software.

**Reply by pauljackman2023** [reply]
I am playing with data from A Sign In Space - Artificially produced "Alien" Radio Signal, 4gb files, X axis and Y axis as if recorded from radio telescope. The files are sigmf-data.

**Reply by cpoore1** [reply]
I am not familiar with radio telescope data. If it has to be handled differently than normal IQ data and you find any tools or algorithms for reading or displaying it, I can add such things to FISSURE to help people out. I'll add something to the IQ Data tab that chops up a huge file into any number of smaller pieces so files can be a little more manageable. Someone else is going to need to comment on this until I can read up on it.

**Reply by pauljackman2023** [reply]
that's great i'm also reading up on how to analyse raw telescope data to see if theres anything i can glean.
